capsules/src/usb_user.rs

```diff
@@ -0,0 +1,137 @@
+//! USB system call interface
```

Comment:
Ok, I can cut out the syscalls and make that part of the next PR if you prefer.

And actually yes I think I'll leave off Hail because it may lead to interesting clock-management issues.

For the utility, I've got a simple one I can just drop in today that gets a few bytes into and out of the host, so I might as well pop that in.